lib/git_service/branch.rb

```diff
@@ -65,6 +65,28 @@ def commit_ids_since(starting_point)
       range_walker(starting_point).collect(&:oid).reverse
     end
 
+    def commit_message(ref)
+      commit = Rugged::Commit.lookup(rugged_repo, ref)
+      parents = commit.parent_oids
+      message = "commit #{commit.oid}\n"
+      message << "Merge: #{parents.join(" ")}\n" if parents.length > 1
+      message << "Author:     #{commit.author[:name]} <#{commit.author[:email]}>\n"
+      message << "AuthorDate: #{commit.author[:time].to_time.strftime("%c %z")}>\n"
+      message << "Commit:     #{commit.author[:name]} <#{commit.author[:email]}>\n"
+      message << "CommitDate: #{commit.author[:time].to_time.strftime("%c %z")}>\n"
+      message << "\n"
+      commit.message.each_line { |line| message << "    #{line}"}
+      message << "\n"
+      previous_commit = Rugged::Commit.lookup(rugged_repo, commit.parent_oids.first)
+      diff = previous_commit.diff(commit)
+      our_diff = Diff.new(diff)
+      our_diff.file_status.each do |file, stats|
+        message << " #{file} | #{stats[:changes]} #{"+" * stats[:additions]}#{"-" * stats[:deletions]}\n"
```

Comment:
Need to deal with large number of changes...they do some kind of math where there is a max number of sigils, then everything is relative to that.

Also the file name is abbreviated in an interesting way if the name is too long.  It basically chops off prefix paths with `...` until it fits some common width

Reply:
yeah... do you think we need to do this now?